File: gitlfs-client/src/test/java/ru/bozaro/gitlfs/client/YamlHelper.java
package ru.bozaro.gitlfs.client;

import org.yaml.snakeyaml.DumperOptions;
import org.yaml.snakeyaml.Yaml;
import org.yaml.snakeyaml.introspector.BeanAccess;

import javax.annotation.Nonnull;
import java.util.ArrayList;
import java.util.List;

/**
 * Helper for YAML.
 *
 * @author Artem V. Navrotskiy
 */
public class YamlHelper {
  @Nonnull
  private static final Yaml YAML = createYaml();

  @Nonnull
  private static Yaml createYaml() {
    final DumperOptions options = new DumperOptions();
    options.setLineBreak(DumperOptions.LineBreak.UNIX);
    options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
    final Yaml yaml = new Yaml(new YamlConstructor(), new YamlRepresenter(), options);
    yaml.setBeanAccess(BeanAccess.FIELD);
    return yaml;
  }

  @Nonnull
  public static HttpReplay createReplay(@Nonnull String resource) {
    final List<HttpRecord> records = new ArrayList<>();
    for (Object item : YamlHelper.get().loadAll(YamlHelper.class.getResourceAsStream(resource))) {
      records.add((HttpRecord) item);
    }
    return new HttpReplay(records);
  }

  @Nonnull
  public static Yaml get() {
    return YAML;
  }
}

File: gitlfs-client/src/test/kotlin/ru/bozaro/gitlfs/client/YamlHelper.kt
package ru.bozaro.gitlfs.client

import org.yaml.snakeyaml.DumperOptions
import org.yaml.snakeyaml.Yaml
import org.yaml.snakeyaml.introspector.BeanAccess

/**
 * Helper for YAML.
 *
 * @author Artem V. Navrotskiy
 */
object YamlHelper {
    private val YAML = createYaml()

    private fun createYaml(): Yaml {
        val options = DumperOptions()
        options.lineBreak = DumperOptions.LineBreak.UNIX
        options.defaultFlowStyle = DumperOptions.FlowStyle.BLOCK
        val yaml = Yaml(YamlConstructor(), YamlRepresenter(), options)
        yaml.setBeanAccess(BeanAccess.FIELD)
        return yaml
    }

    fun createReplay(resource: String): HttpReplay {
        val records: MutableList<HttpRecord> = ArrayList()
        for (item in get().loadAll(YamlHelper::class.java.getResourceAsStream(resource))) {
            records.add(item as HttpRecord)
        }
        return HttpReplay(records)
    }

    fun get(): Yaml {
        return YAML
    }
}
